force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,48 @@
+trigger AccountTrigger on Account (before insert, before update, after insert) {
+    List<Contact> contactsToInsert = new List<Contact>();
+
+    // Set the account type to prospect.
+    if (Trigger.isBefore) {
+        for (Account account : Trigger.New) {
+            // Set the account type to "Prospect" if not set
+            if (account.Type == null) {
+                account.Type = 'Prospect';
+            }
+
+            // Copy the shipping address to the billing address.
+            if (Trigger.isInsert) {
+                if (account.ShippingStreet != null) {
```

Comment:
I would check that the target is not blank, not the source. It does not matter if you are copying over blank fields, but you might not want to overwrite values if they are supplied already in Billing address values.